pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,639 @@
+	inUseClaim = st.FromResourceClaim(pendingImmediateClaim).
+			Allocation(&resourcev1alpha1.AllocationResult{}).
+			ReservedFor(resourcev1alpha1.ResourceClaimConsumerReference{UID: types.UID(podUID)}).
+			Obj()
+	allocatedClaim = st.FromResourceClaim(pendingDelayedClaim).
+			OwnerReference(podName, podUID, podKind).
+			Allocation(&resourcev1alpha1.AllocationResult{}).
+			Obj()
+	allocatedClaimWithWrongTopology = st.FromResourceClaim(allocatedClaim).
+					Allocation(&resourcev1alpha1.AllocationResult{AvailableOnNodes: st.MakeNodeSelector().In("no-such-label", []string{"no-such-value"}).Obj()}).
+					Obj()
+	allocatedClaimWithGoodTopology = st.FromResourceClaim(allocatedClaim).
+					Allocation(&resourcev1alpha1.AllocationResult{AvailableOnNodes: st.MakeNodeSelector().In("nodename", []string{"worker"}).Obj()}).
+					Obj()
+	otherClaim = st.MakeResourceClaim().
+			Name("not-my-claim").
+			Namespace(namespace).
+			ResourceClassName(className).
+			Obj()
+
+	scheduling = st.MakePodScheduling().Name(podName).Namespace(namespace).
+			OwnerReference(podName, podUID, podKind).
+			Obj()
+	schedulingPotential = st.FromPodScheduling(scheduling).
+				PotentialNodes(workerNode.Name).
+				Obj()
+	schedulingSelectedPotential = st.FromPodScheduling(schedulingPotential).
+					SelectedNode(workerNode.Name).
+					Obj()
+	schedulingInfo = st.FromPodScheduling(schedulingPotential).
+			ResourceClaims(resourcev1alpha1.ResourceClaimSchedulingStatus{Name: resourceName},
+			resourcev1alpha1.ResourceClaimSchedulingStatus{Name: resourceName2}).
+		Obj()
+)
+
+// result defines the expected outcome of some operation. It covers
+// operation's status and the state of the world (= objects).
+type result struct {
+	status *framework.Status
+
+	// changes contains a mapping of name to an update function for
+	// the corresponding object. These functions apply exactly the expected
+	// changes to a copy of the object as it existed before the operation.
+	changes changeMapping
+
+	// added contains objects created by the operation.
+	added []metav1.Object
+
+	// removed contains objects deleted by the operation.
+	removed []metav1.Object
+}
+
+type changeMapping map[string]func(metav1.Object) metav1.Object
+type perNodeResult map[string]result
+
+func (p perNodeResult) forNode(nodeName string) result {
+	if p == nil {
+		return result{}
+	}
+	return p[nodeName]
+}
+
+type want struct {
+	preFilterResult *framework.PreFilterResult
+	prefilter       result
+	filter          perNodeResult
+	prescore        result
+	reserve         result
+	unreserve       result
+	postbind        result
+}
+
+// prepare contains changes for objects in the API server.
+// Those changes are applied before running the steps. This can
+// be used to simulate concurrent changes by some other entities
+// like a resource driver.
+type prepare struct {
+	filter    changeMapping
+	prescore  changeMapping
+	reserve   changeMapping
+	unreserve changeMapping
+	postbind  changeMapping
```

Comment:
and postfilter?

Reply:
I started working on it today, but haven't finished it because it wasn't clear under which circumstances the framework would call it.

Reply:
This is a gap, and it is better to address it now. I finished the unit tests and will push them soon.

Reply:
Done.

I am also including some other unit test enhancement (checking for conflicts), all in separate commits that could, but don't have to be squashed.